### PR TITLE
bundleid peek queryparam

### PR DESCRIPTION
**Pull request by xnetsrak**
## Description

Added bundleId to peek query param as required 
## References

* #183

### CODE REVIEW

File: source/Energinet.DataHub.PostOffice.Application/Handlers/PeekHandler.cs [hunk 1]
```diff
@@ -38,7 +38,7 @@ public async Task<PeekResponse> Handle(PeekCommand request, CancellationToken ca
                 throw new ArgumentNullException(nameof(request));
 
             var bundle = await _marketOperatorDataDomainService
-                .GetNextUnacknowledgedAsync(new MarketOperator(new GlobalLocationNumber(request.Recipient)))
+                .GetNextUnacknowledgedAsync(new MarketOperator(new GlobalLocationNumber(request.Recipient)), new Uuid(request.BundleId))
```
Comment (defectiveAi) [inline source/Energinet.DataHub.PostOffice.Application/Handlers/PeekHandler.cs:41]: I can see the bundleId is a guid today. But is that still a good thing, when we ask the market operators to provide it? Maybe the bundle id should just be a string, and the market operator can put whatever unique value they want. Like the example Søren used, where market operators have a sequential integer.

Comment (xnetsrak) [reply source/Energinet.DataHub.PostOffice.Application/Handlers/PeekHandler.cs:41]: What happens if the market operators provide the same bundleid ?  that could happen a lot if the the bundleid is only a integer.

Comment (defectiveAi) [reply source/Energinet.DataHub.PostOffice.Application/Handlers/PeekHandler.cs:41]: We have to handle it somehow. MO can provide the same Guid multiple times, which is also illegal.

Comment (FirestarJes) [reply source/Energinet.DataHub.PostOffice.Application/Handlers/PeekHandler.cs:41]: I agree with @defectiveAi on this one, and that we need to handle it.

File: source/Energinet.DataHub.PostOffice.Domain/Model/BundleCreatedResponse.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Energinet.DataHub.PostOffice.Domain.Model
+{
+    public enum BundleCreatedResponse
```
Comment (defectiveAi) [inline source/Energinet.DataHub.PostOffice.Domain/Model/BundleCreatedResponse.cs:17]: This is just a suggestion, so pls ignore. But a more domain-oriented names could be:
- Success
- AnotherBundleExists
- BundleIdAlreadyInUse

Or some variant. Errors are more for severe stuff, and should probably be exceptions. These are more "domain expected" circumstances that we have to deal with.

File: source/Energinet.DataHub.PostOffice.Infrastructure/Mappers/BundleMapper.cs [hunk 3]
```diff
@@ -30,10 +30,16 @@ public static BundleDocument MapToDocument(Bundle source)
...
                 Dequeued = false,
-                ContentPath = MapBundleContent(source)
+                ContentPath = MapBundleContent(source),
+                ProcessId = GenerateProcessId(source)
             };
         }
 
+        private static string GenerateProcessId(Bundle bundle)
+        {
+            return string.Concat(bundle.BundleId.ToString(), bundle.Recipient.Gln.Value);
```
Comment (defectiveAi) [inline source/Energinet.DataHub.PostOffice.Infrastructure/Mappers/BundleMapper.cs:40]: Add a separator between the values, so we can separate the programmatically down the line. Something like +

File: source/Energinet.DataHub.PostOffice.IntegrationTests/Repositories/BundleRepositoryIntegrationTests.cs [hunk 4]
```diff
@@ -162,12 +162,12 @@ public async Task TryAddNextUnacknowledgedAsync_NoExistingBundle_ReturnsTrue()
             var couldAdd = await target.TryAddNextUnacknowledgedAsync(setupBundle).ConfigureAwait(false);
 
             // Assert
-            Assert.True(couldAdd);
+            Assert.True(couldAdd == BundleCreatedResponse.Success);
```
Comment (defectiveAi) [inline source/Energinet.DataHub.PostOffice.IntegrationTests/Repositories/BundleRepositoryIntegrationTests.cs:165]: If you use Equals, we get pretty formatted message when the test fails.

File: source/Energinet.DataHub.PostOffice.Application/Handlers/PeekHandler.cs (hunk 1 above)
Comment (defectiveAi) [reply source/Energinet.DataHub.PostOffice.Application/Handlers/PeekHandler.cs:41]: Is fixed now

File: source/Energinet.DataHub.PostOffice.Application/Validation/PeekCommandRuleSet.cs [hunk 5]
```diff
@@ -25,6 +25,10 @@ public PeekCommandRuleSet()
             RuleFor(command => command.Recipient)
                 .NotEmpty()
                 .SetValidator(new GlobalLocationNumberValidationRule());
+
+            RuleFor(command => command.BundleId)
+                .NotEmpty()
+                .SetValidator(new UuidValidationRule());
```
Comment (defectiveAi) [inline source/Energinet.DataHub.PostOffice.Application/Validation/PeekCommandRuleSet.cs:31]: Add same rule to PeekCommand for TimeSeries/Aggregations

**Review by defectiveAi — APPROVED** [review]

**Review by michaelmajgaard — APPROVED** [review]

**Pull request merged**